### ISSUE TITLE
Collation parsing problem in spellcheck

### ISSUE THREAD

**Issue by Edminsson** [SolrNet/SolrNet#310]
In SolrNet 5.x when spellcheck is enabled and spellcheck.collateExtendedResults is false then the result looks like this
```   
<lst name="spellcheck">
    <lst name="suggestions"/>
    <bool name="correctlySpelled">false</bool>
    <lst name="collations">
        <str name="collation">animal</str>
    </lst>
</lst>
 ```
The collation result is not parsed correctly because the element with name="collation" is not an lst element but an str element.

**Comment by xavier-morera** [comment]
Thanks. I need this one as well.

**Comment by Edminsson** [comment]
Hi @xavier-morera, I'll give it a try.

**Comment by xavier-morera** [comment]
Excellent! I created a branch for this purpose: https://github.com/SolrNet/SolrNet/tree/collation-parsing-issue-310?files=1

Thank you

**Comment by Edminsson** [comment]
I also found a problem with the parsing av suggestions. I think I fixed both problems. It's been a long time since I worked with Solr and I tried to run all the tests but I had some trouble running some of them. For instance the NHibernate tests seem to need an SQLite driver.

**Comment by Edminsson** [comment]
There seems to be the same issue with suggestions and spellcheck.extendedResults as with collations and  spellcheck.collateExtendedResults. I'll fix the code.

**Comment by xavier-morera** [comment]
Thanks @Edminsson

**Comment by gjunge** [comment]
This branch has to be updated, it cannot merge anymore with Develop

**Comment by gjunge** [comment]
Fixed, will be part of the 1.0.0 release.
